### PR TITLE
Update 'transaction' jcli command, add posibility to create evm transaction fragment

### PR DESCRIPTION
**Pull request by Mr-Leshiy**
Firstly need to merge https://github.com/input-output-hk/chain-libs/pull/726

Updated `Transaction` jcli command, add new subcommand `AddEvmTransaction` for the `Transaction` command.
Add `EvmTransaction` interface type for the library's `EvmTransaction` type

### CODE REVIEW

**Review by enzoc4 — COMMENTED** [review]
why can't the `EvmTransaction` be a certificate? Maybe it was discussed already, but doesn't it follow the same pattern?

**Comment by Mr-Leshiy** [comment]
@enzoc4 it is a good question. Indeed currently we do not have any cryptographic validation of the evm transactions, which is a vulnerability.
I am not sure but assume that it is not implemented currently  because we also need to determine how to sing such transactions and how to match our `jormungandr` address format with the `evm` address format and now we are trying to add some MVP part of the evm.

@eugene-babichenko maybe can elaborate about it.

**Comment by Mr-Leshiy** [comment]
Also needs to merge first this https://github.com/input-output-hk/jormungandr/pull/3757

File: jcli/src/jcli_lib/transaction/mod.rs [hunk 1]
```diff
@@ -45,9 +46,15 @@ pub enum Transaction {
     /// set a transaction expiration date
     SetExpiryDate(set_expiry_date::SetExpiryDate),
     /// set a certificate to the Transaction. If there is already
-    /// an extra certificate in the transaction it will be replaced
-    /// with the new one.
+    /// an evm transaction in the transaction it will be reseted.
```
Comment (eugene-babichenko) [inline jcli/src/jcli_lib/transaction/mod.rs:49]:
```suggestion
    /// an evm transaction in the transaction it will be reset.
```

File: jcli/src/jcli_lib/transaction/mod.rs [hunk 2]
```diff
@@ -45,9 +46,15 @@ pub enum Transaction {
...
+    /// If there is already an extra certificate in the transaction
+    /// it will be replaced with the new one.
     AddCertificate(add_certificate::AddCertificate),
+    /// set a evm transaction to the Transaction. If there is already
+    /// an extra certificate in the transaction it will be reseted.
```
Comment (eugene-babichenko) [inline jcli/src/jcli_lib/transaction/mod.rs:54]:
```suggestion
    /// an extra certificate in the transaction it will be reset.
```

File: jcli/src/jcli_lib/transaction/staging.rs [hunk 3]
```diff
@@ -358,10 +375,19 @@ impl Staging {
...
+            None => match &self.evm_transaction {
+                None => self.finalize_payload(
+                    &chain::transaction::NoExtra,
+                    fee_algorithm,
+                    output_policy,
+                ),
+                Some(_tx) => {
+                    // let tx: chain::evm::EvmTransaction = tx.clone().into();
+                    // self.finalize_payload(&tx, fee_algorithm, output_policy)
+                    unreachable!()
```
Comment (eugene-babichenko) [inline jcli/src/jcli_lib/transaction/staging.rs:387]: what's the reason for this change?

File: jcli/src/jcli_lib/transaction/staging.rs [hunk 4]
```diff
@@ -573,8 +606,15 @@ impl Staging {
...
         let res = match &self.extra {
-            None => self.transaction_sign_data_hash_on(TxBuilder::new().set_nopayload()),
-            Some(ref c) => match c.clone().into() {
+            None => match &self.evm_transaction {
+                None => self.transaction_sign_data_hash_on(TxBuilder::new().set_nopayload()),
+                Some(_tx) => {
+                    // let tx: chain::evm::EvmTransaction = tx.clone().into();
+                    // self.transaction_sign_data_hash_on(TxBuilder::new().set_payload(&tx))
+                    unreachable!()
+                }
```
Comment (eugene-babichenko) [inline jcli/src/jcli_lib/transaction/staging.rs:615]: Here's another one

File: jcli/src/jcli_lib/transaction/staging.rs [hunk 5]
```diff
@@ -358,10 +375,21 @@ impl Staging {
...
+                    output_policy,
+                ),
+                Some(tx) => {
+                    #[allow(unreachable_code)]
+                    #[allow(unused_variables)]
+                    let tx: chain::evm::EvmTransaction = tx.clone().into();
+                    #[allow(unreachable_code)]
+                    self.finalize_payload(&tx, fee_algorithm, output_policy)
+                }
+            },
```
Comment (eugene-babichenko) [inline jcli/src/jcli_lib/transaction/staging.rs:391]: Are these really unreachable/unused now?

Comment (Mr-Leshiy) [reply jcli/src/jcli_lib/transaction/staging.rs:391]: We have the following situation at this place. When we are compiling with the `feature = "evm"` so it works fine and compiles with any problem, but when we are compiling without  `feature = "evm"` so `chain::evm::EvmTransaction` become a `uninhabited` type, so it is not way how to initialize a value with this type so according to this this code will be unreachable.
```
pub enum EvmTransaction {}
```

Maybe it is better than I will do it in the following way
```
#[cfg(feature(evm)]
{
     let tx: chain::evm::EvmTransaction = tx.clone().into();
     self.finalize_payload(&tx, fee_algorithm, output_policy)
}
#[cfg(feature(evm)]
unreachable!()
```

Comment (eugene-babichenko) [reply jcli/src/jcli_lib/transaction/staging.rs]: That would be the correct approach IMO. These unreachables would hit us in the back later.

**Review by eugene-babichenko — APPROVED** [review]
